tensorflow/core/kernels/conv_ops_using_gemm.cc

```diff
@@ -262,14 +262,15 @@ class Im2ColConvFunctor {
                 errors::InvalidArgument("Im2Col patch too large for buffer"));
     const size_t patches_per_chunk =
         max_chunk_size / (filter_value_count * sizeof(T1));
+    const size_t chunk_value_count = max_chunk_size / sizeof(T1);
```

Comment:
This is a bug fix because I was actually using max_chunk_size (which is in bytes) as the number of _elements_ to allocate below. This led to a 64MB buffer being allocated for 16m floats, rather than the 16MB buffer with 4m floats that I was expecting.

Good point on the size calculation! I've updated it to effective do a ceil on the integer calculation, to make sure we allocate enough space even for odd-sized types.